pkg/storage/s3/s3.go

```diff
@@ -523,6 +547,16 @@ func (d *driver) RemoveStorage(cr *imageregistryv1.Config) (bool, error) {
 	if err != nil {
 		return false, err
 	}
+
+	iter := s3manager.NewDeleteListIterator(svc, &s3.ListObjectsInput{
+		Bucket: aws.String(d.Config.Bucket),
+	})
+
+	err = s3manager.NewBatchDeleteWithClient(svc).Delete(aws.BackgroundContext(), iter)
+	if err != nil && !isBucketNotFound(err) {
+		return false, err
+	}
```

Comment:
For below - what if we try to delete a bucket that doesn't exist? Feels like we shouldn't report an error in that case.

Reply:
We don't report an error, we update the condition to "S3 Bucket Deleted" and return.